app/src/main/java/com/framgia/wsm/screen/requestoff/RequestOffViewModel.java

```diff
@@ -685,77 +685,243 @@ private double getSumDateOffHaveSalary() {
+                    mRequestOff.getMiscarriageLeave())) {
+                mRequestDayOffTypesAttribute.setDestroy(TRUE);
+            }
         }
-        if (mRequestOff.getMiscarriageLeave() != null && !"".equals(
-                mRequestOff.getMiscarriageLeave())) {
-            mRequestDayOffTypesAttribute.setSpecialDayOffSettingId(TypeOfDays.MISCARRIAGE_LEAVE);
-            mRequestDayOffTypesAttribute.setNumberDayOff(mRequestOff.getMiscarriageLeave());
-            mRequestDayOffTypesAttribute.setDestroy(FALSE);
-            mRequestDayOffTypesAttributes.add(mRequestDayOffTypesAttribute);
+        mRequestDayOffTypesAttributes.add(mRequestDayOffTypesAttribute);
+    }
+
+    private void setAttributeSickLeave() {
+        mRequestDayOffTypesAttribute = new OffRequest.RequestDayOffTypesAttribute();
+        mRequestDayOffTypesAttribute.setId(null);
+        mRequestDayOffTypesAttribute.setSpecialDayOffSettingId(TypeOfDays.SICK_LEAVE);
+        mRequestDayOffTypesAttribute.setNumberDayOff(mRequestOff.getSickLeave());
+        mRequestDayOffTypesAttribute.setDestroy(FALSE);
+        if (mRequestOff != null && (mRequestOff.getRequestDayOffTypes() != null
+                || mRequestOff.getRequestDayOffTypes().size() != 0)) {
+            for (int i = 0; i < mRequestOff.getRequestDayOffTypes().size(); i++) {
+                if (mRequestOff.getRequestDayOffTypes().get(i).getSpecialDayOffSettingId()
+                        == Integer.parseInt(TypeOfDays.SICK_LEAVE)) {
+                    mRequestDayOffTypesAttribute.setId(
+                            mRequestOff.getRequestDayOffTypes().get(i).getId());
+                }
+            }
+            if (mRequestOff.getSickLeave() == null
+                    || "".equals(mRequestOff.getSickLeave())
+                    || Constant.DEFAULT_DOUBLE_VALUE.equals(mRequestOff.getSickLeave())) {
+                mRequestDayOffTypesAttribute.setDestroy(TRUE);
+            }
         }
-        if (mRequestOff.getSickLeave() != null && !"".equals(mRequestOff.getSickLeave())) {
-            mRequestDayOffTypesAttribute.setSpecialDayOffSettingId(TypeOfDays.SICK_LEAVE);
-            mRequestDayOffTypesAttribute.setNumberDayOff(mRequestOff.getSickLeave());
-            mRequestDayOffTypesAttribute.setDestroy(FALSE);
-            mRequestDayOffTypesAttributes.add(mRequestDayOffTypesAttribute);
+        mRequestDayOffTypesAttributes.add(mRequestDayOffTypesAttribute);
+    }
+
+    private void setAttributePregnancyExaminationLeave() {
+        mRequestDayOffTypesAttribute = new OffRequest.RequestDayOffTypesAttribute();
+        mRequestDayOffTypesAttribute.setId(null);
+        mRequestDayOffTypesAttribute.setSpecialDayOffSettingId(TypeOfDays.PREGNANCY_EXAMINATON);
+        mRequestDayOffTypesAttribute.setNumberDayOff(mRequestOff.getPregnancyExaminationLeave());
+        mRequestDayOffTypesAttribute.setDestroy(FALSE);
+        if (mRequestOff != null && (mRequestOff.getRequestDayOffTypes() != null
```

Comment:
ERROR: [BrokenNullCheck] Method call on object which may be null
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/basic.html#BrokenNullCheck